testing/integration/fqdn_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestFQDN(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Stack: &define.Stack{},
+		Local: false,
+	})
+
+	suite.Run(t, &FQDN{requirementsInfo: info})
+}
+
+type FQDN struct {
+	suite.Suite
+	requirementsInfo *define.Info
+	agentFixture     *atesting.Fixture
+
+	externalIP string
+	origFQDN   string
+}
+
+// Before suite
+func (s *FQDN) SetupSuite() {
+	agentFixture, err := define.NewFixture(s.T())
+	require.NoError(s.T(), err)
+	s.agentFixture = agentFixture
+
+	externalIP, err := getExternalIP()
+	require.NoError(s.T(), err)
+	s.externalIP = externalIP
+}
+
+// Before each test
+func (s *FQDN) SetupTest() {
+	ctx := context.Background()
+
+	// Save original hostname so we can restore it at the end of each test
+	origFQDN, err := getHostFQDN(ctx)
+	require.NoError(s.T(), err)
+	s.origFQDN = origFQDN
+}
+
+func (s *FQDN) TearDownTest() {
+	ctx := context.Background()
+
+	// Restore original FQDN
+	err := setHostFQDN(ctx, s.externalIP, s.origFQDN)
+	require.NoError(s.T(), err)
+}
+
+func (s *FQDN) TestFQDN() {
+	ctx := context.Background()
+	kibClient := s.requirementsInfo.KibanaClient
+
+	// Set FQDN on host
+	shortName := randStr(6)
+	fqdn := shortName + ".baz.io"
+	err := setHostFQDN(ctx, s.externalIP, fqdn)
+	require.NoError(s.T(), err)
+
+	// Create Agent policy
+	createPolicyReq := kibana.CreatePolicyRequest{
+		Name:        "test-policy-fqdn-" + strings.ReplaceAll(fqdn, ".", "-"),
+		Namespace:   "default",
+		Description: fmt.Sprintf("Test policy for FQDN E2E test (%s)", fqdn),
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policy, err := kibClient.CreatePolicy(createPolicyReq)
+	require.NoError(s.T(), err)
+
+	// Create enrollment API key
+	createEnrollmentApiKeyReq := kibana.CreateEnrollmentAPIKeyRequest{
+		PolicyID: policy.ID,
+	}
+	enrollmentToken, err := kibClient.CreateEnrollmentAPIKey(createEnrollmentApiKeyReq)
+	require.Nil(s.T(), err, "Could not create enrollment token")
+
+	// Get default Fleet Server URL
+	fleetServerURL, err := tools.GetDefaultFleetServerURL(kibClient)
+	require.NoError(s.T(), err)
```

Comment:
I think that this section (create policy, create enrollment api key, and getting the fleet server) would be great in tools as a simple function to get all this information in a single call in a test. I feel like lots of tests will be doing this.

Reply:
Extracted this section out to tools in 49e9129c2.